commit 51aa32c32eb72dc56127daac797a890a85d47823 (refs/changes/55/3155/1)
Author: Kelly McLaughlin <kelly.mclaughlin@joyent.com>
Date:   2018-01-04T11:49:15-07:00 (1 year, 9 months ago)
    
    MANTA-3511 Repeated PUT requests with If-Match: "" result in socket hangs or hangups

diff --git a/lib/server.js b/lib/server.js
index 3e3622e..624dd9e 100644
--- a/lib/server.js
+++ b/lib/server.js
@@ -461,7 +461,7 @@ function createServer(options, clients, name) {
         _audit(req, res, route, err);
 
         if ((req.method === 'PUT' || req.method === 'POST') &&
-            res.statusCode >= 500) {
+            res.statusCode >= 400) {
             /*
              * An error occurred on a PUT or POST request, but there may still
              * be incoming data on the request stream. Call resume() in order to
